comment/managers/blocker.py

```diff
@@ -0,0 +1,40 @@
+from django.db import models
+
+from comment.conf import settings
+
+
+class BlockedUserManager(models.Manager):
+    def is_user_blocked(self, user_id=None, email=None):
+        if not settings.COMMENT_ALLOW_BLOCKING_USERS:
+            return False
+        if user_id:
+            return self.is_user_blocked_by_id(user_id)
+        elif email:
+            return self.is_user_blocked_by_email(email)
+        return False
```

Comment:
you can probably save some queries to the database in some scenarios with the following:

```python
        if not user_id and not email:
            return False
        elif user_id:
            return self.is_user_blocked_by_id(user_id)
        elif email:
            return self.is_user_blocked_by_email(email)
```

Reply:
I am not sure what you refer to but the code above won't hit the DB unless the email or user_id is provided.